### DISCUSSION TITLE
Why does JWT middleware throw "Error: invalid credentials structure"

### DISCUSSION THREAD

**Discussion by Swarz** [honojs/hono#2978] [Q&A] [answered]
Simulated jwt and specified the Authorization header value, but it did not pass the verification.

1. decode info
2. throw  error
![微信截图_20240616091555](https://github.com/honojs/hono/assets/38528567/14e8685f-1e89-4e6a-a578-5ba379ea5c9a)

how to verify through JWT middleware ?

**Comment by NicoPlyley** [comment, marked as answer]
Hi @Swarz 

This error is likely because you are sending the token without prepending the type of token. Generally bearer is used, so you would set your authorization to `Bearer my_jwt_token`

**Reply by Swarz** [reply]
thx, 
then i have read line 14 of  the jwt file  (\hono\dist\middleware\jwt\jwt.js) to confirm that

```
    const credentials = ctx.req.raw.headers.get("Authorization");
    let token;
    if (credentials) {
      const parts = credentials.split(/\s+/);
      if (parts.length !== 2) {
        const errDescription = "invalid credentials structure";
        throw new HTTPException(401, {
          message: errDescription,
          res: unauthorizedResponse({
            ctx,
            error: "invalid_request",
            errDescription
          })
        });
      }
...
```
